openrr-base/src/odometry.rs

```diff
@@ -0,0 +1,109 @@
+use arci::{BaseVelocity, Isometry2, Vector2};
+use parking_lot::Mutex;
+
+use crate::Error;
+
+const ODOMETRY_TIMEOUT_MILLIS: u128 = 5000;
+
+#[derive(Debug)]
+pub struct Odometry {
+    /// If `None`, it is odometry lost.
+    position: Mutex<Option<Isometry2<f64>>>,
+    last_update_timestamp: Mutex<Option<std::time::Instant>>,
+}
+
+impl Odometry {
+    pub fn new_from_position_with_now(position: Isometry2<f64>) -> Self {
+        Self {
+            position: Mutex::new(Some(position)),
+            last_update_timestamp: Mutex::new(None),
+        }
+    }
+
+    pub fn translate(&self, velocity: &BaseVelocity) -> Result<(), Error> {
```

Comment:
translate means move only position. This contains rotation.
So it's better to avoid the name.
My suggestion is `update_by_velocity()`, `transform()`.